src/tools/interop/idt/capture/base.py

```diff
@@ -0,0 +1,92 @@
+#
+#    Copyright (c) 2023 Project CHIP Authors
+#    All rights reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+
+from abc import ABC, abstractmethod
+
+
+class PlatformLogStreamer(ABC):
+    """
+    The abstract base class for a platform transport, subclassed by sub packages of platform
+    """
+
+    @abstractmethod
+    def __init__(self, artifact_dir: str) -> None:
+        """
+         artifact_dir: the fully qualified path of the output directory. This directory already exists
+        """
```

Comment:
i'm no python expert, so i'm probably wrong, but wouldn't something normally complain about an empty method here? ie, do you need a "pass"?

Reply:
Thanks! I've actually moved them all to `raise NotImplementedError`. 

Some context on conventions:

https://github.com/python/cpython/issues/100400